src/main/java/edu/hawaii/its/api/service/GroupingAssignmentService.java

```diff
@@ -120,14 +123,17 @@ public GroupingGroupMembers groupingOwners(String currentUser, String groupingPa
                 grouperService.getMembersResult(currentUser, groupingPath + GroupType.OWNERS.value()));
     }
 
-    public Boolean isSoleOwner(String currentUser, String groupPath, String uidToCheck) {
+    public Integer numberOfOwners(String currentUser, String groupPath, String uhIdentifier) {
         logger.debug(String.format("isSoleOwner; currentUser: %s; groupPath: %s; uidToCheck: %s",
-                currentUser, groupPath, uidToCheck));
+                currentUser, groupPath, uhIdentifier))  ;
+        if (!memberService.isOwner(uhIdentifier)) {
+            throw new AccessDeniedException();
+        }
         List<GroupingGroupMember> owners = groupingOwners(currentUser, groupPath).getMembers();
-        if (owners.size() >= 2) {
-            return false;
+        if (owners.stream().noneMatch(owner -> owner.getUid().contains(uhIdentifier))) {
+            throw new UhMemberNotFoundException("Owner " + uhIdentifier + " not found");
```

Comment:
Ok let's go forward with this PR, there may be a use for getting the number of owners in the future. 

This stream check can be removed, the isOwner check above performs the same check.

Reply:
Okay! Sounds good